app/models/product_import/spreadsheet_entry.rb

```diff
@@ -92,5 +96,10 @@ def non_product_attributes
        'product_validations', 'inventory_validations', 'validates_as',
        'save_type', 'on_hand_nil', 'has_overrides']
     end
+
+    def non_assignable_attributes
+      ['producer', 'producer_id', 'category', 'shipping_category', 'tax_category',
+       'units', 'unscaled_units', 'unit_type', 'enterprise', 'enterprise_id']
+    end
```

Comment:
I would rather move this array to a constant (and call #freeze on it so we forbid mutation). This way Ruby evaluates at once at boot time and not in every request. The frozen string comment already avoids the string allocations on every request so there's less gain with the constant though.

Reply:
Done.

Reply:
Also rebased to fix the build...